main/src/cgeo/geocaching/CacheListActivity.java

```diff
@@ -1133,6 +1123,20 @@ private String getDisplayName(final Uri uri) {
     }
 
     public void refreshStored(final List<Geocache> caches) {
+        if (type == CacheListType.OFFLINE && caches.size() > REFRESH_WARNING_THRESHOLD) {
+            Dialogs.confirmYesNo(this, R.string.caches_refresh_all, R.string.caches_refresh_all_warning, new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(final DialogInterface dialog, final int id) {
+                    refreshStoredConfirmed(caches);
+                    dialog.cancel();
```

Comment:
Interesting question. I copied the Dialog code from somewhere else in cgeo without fully understanding it.

According to:
https://developer.android.com/guide/topics/ui/dialogs.html#DismissingADialog
> When the user touches any of the action buttons created with an AlertDialog.Builder, the system dismisses the dialog for you.

So is this extra `dialog.cancel()` necessary at all? I don't know the exact lifecycle of the AlertDialog. Maybe the automatic closing is done after `onClick()` has been called. Then it might be necessary to cancel the dialog before continuing the processing to prevent any further Dialogs to stack on top of it.

Should I put it before `refreshStoredConfirmed(caches);`, to be on the safe side?